### ISSUE TITLE
Liste des tags disponibles pour les Pawns

### ISSUE THREAD

**Issue by b606** [Ludeon/RimWorld-fr#293]
Pour mémoire, les tags disponibles pour les Pawns:
[List of available symbols for Pawns](https://ludeon.com/forums/index.php?topic=43979.0#new)
`{0}`: John, a cat, a dog (short name if possible, otherwise indefinite form of the pawn kind)
`{0_nameFull}`: John Doe, a cat, a dog (full name if possible, otherwise indefinite form of the pawn kind)
`{0_nameFullDef}`: John Doe, the cat, the dog (full name if possible, otherwise definite form of the pawn kind)
`{0_label}`: John, Constructor; cat; dog (label (name + title))
`{0_labelShort}`: John, cat, dog (short name if possible, otherwise kind label)
`{0_definite}`: John, the cat, the dog (short name if possible, otherwise definite form of the pawn kind)
`{0_nameDef}`: John, the cat, the dog (short name if possible, otherwise definite form of the pawn kind)
`{0_indefinite}`: John, a cat, a dog (short name if possible, otherwise indefinite form of the pawn kind)
`{0_nameIndef}`: John, a cat, a dog (short name if possible, otherwise indefinite form of the pawn kind)
`{0_pronoun}`: he/she, en français **il/elle**
`{0_possessive}`: his/her, en français **son/sa**
`{0_objective}`: him/her, en français **lui/elle**
`{0_factionName}`: Some Community (faction name)
`{0_factionPawnSingular}`: colonist (faction member label)
`{0_factionPawnSingularDef}`: the colonist (faction member label (definite))
`{0_factionPawnSingularIndef}`: a colonist (faction member label (indefinite))
`{0_factionPawnPlural}`: colonists (faction members label)
`{0_kind}`: human, cat, dog (kind label)
`{0_kindDef}`: the human, the cat, the dog (kind label (definite))
`{0_kindIndef}`: a human, a cat, a dog (kind label (indefinite))
`{0_kindPlural}`: humans, cats, dogs (kind label (plural))
`{0_kindBase}`: human, cat, dog, deer (instead of buck/doe) (base genderless kind label)
`{0_kindBaseDef}`: the human, the cat, the dog, the deer (base genderless kind label (definite))
`{0_kindBaseIndef}`: a human, a cat, a dog, a deer (base genderless kind label (indefinite))
`{0_kindBasePlural}`: humans, cats, dogs, deer (base genderless kind label (plural))
`{0_lifeStage}`: teenager, baby, adult (life stage label)
`{0_lifeStageDef}`: the teenager, the baby, the adult (life stage label (definite))
`{0_lifeStageIndef}`: a teenager, a baby, an adult `(life` stage label (indefinite))
`{0_lifeStageAdjective}`: teenage, baby, adult
`{0_title}`: constructor (pawn title)
`{0_titleDef}`: the constructor (pawn title (definite))
`{0_titleIndef}`: a constructor (pawn title (indefinite))
{0_gender}: male, female (pawn gender)
`{0_gender ? A : B}`: A if male or neuter, B if female
`{0_gender ? A : B : C}`: A if male, B if female, C if neuter
`{0_humanlike ? A : B}`: A if humanlike, B if not

EDIT: Notes venant du [thread](https://ludeon.com/forums/index.php?topic=43979.0#new)
- Note that we now use {} everywhere instead of []. The RulePackDef system (used by art descriptions and name generators) still uses [] though. Even though the RulePackDefs system works similarly they are 2 separate systems.
- Our system can easily tell the gender of a pawn, but it can't tell the gender of any other object. This means that if you want to use a gender conditional expression on, for example, items, then you need to provide some lookup tables by creating the following files in your language folder:
WordInfo/Gender/Male.txt
WordInfo/Gender/Female.txt
WordInfo/Gender/Neuter.txt
and add one word per line to each file, e.g.: (Male.txt)

**Comment by kiliak** [comment]
ne les connaissant pas bien pour ma part, je pense qu'il faudrait aussi préciser dans quel type de fichier ils sont untilisable ou non. et voir pour rajouter tout les autres tags.. je sais il y en un paquet.

**Comment by b606** [comment]
Exact. Pour les autres tags, comme il y en a tellement, le mieux à faire c'est d'activer le `log Grammar Resolution` en dev mode, et voir dans le fichier `rimword.log` la liste complète.

En relisant ci-dessus, les problèmes évoqués dans #283 et #284 viennent peut-être de l'utilisation des `{0_kindBase}` (base **genderless** kind label) au lieu de `{0_kind}`. D'où "la bouquetin" au lieu de "l'étagne".

**Comment by b606** [comment]
Avec Royalty, on a les titres royaux en plus (listés dans #288 )

```
  asker_nameFull-><color=#D09B61FF>Pulcheria Tornikos</color>
  asker_label-><color=#D09B61FF>Tornikos</color>
  asker_definite-><color=#D09B61FF>Tornikos</color>
  asker_nameDef-><color=#D09B61FF>Tornikos</color>
  asker_indefinite-><color=#D09B61FF>Tornikos</color>
  asker_nameIndef-><color=#D09B61FF>Tornikos</color>
  asker_pronoun->elle
  asker_possessive->sa
  asker_objective->elle
  asker_factionName-><color=#00BFFFFF>The Exodus Dominion</color>
  asker_kind->gouverneur stellaire
  asker_title->recon sniper
  asker_titleIndef->un recon sniper
  asker_titleDef->le recon sniper
  asker_royalTitle0->gouverneur stellaire
  asker_royalTitle0Indef->un gouverneur stellaire
  asker_royalTitle0Def->le gouverneur stellaire
  asker_royalTitleFaction0-><color=#00BFFFFF>The Exodus Dominion</color>
  asker_royalTitleInCurrentFaction->gouverneur stellaire
  asker_royalTitleInCurrentFactionIndef->un gouverneur stellaire
  asker_royalTitleInCurrentFactionDef->le gouverneur stellaire
  asker_bestRoyalTitle->gouverneur stellaire
  asker_bestRoyalTitleIndef->un gouverneur stellaire
  asker_bestRoyalTitleDef->le gouverneur stellaire
  asker_bestRoyalTitleFaction->The Exodus Dominion
  asker_age->24
  asker_chronologicalAge->647
  asker_relationInfo->
  asker_faction_name-><color=#00BFFFFF>The Exodus Dominion</color>
  asker_faction_pawnsPlural->impériaux
  asker_faction_pawnsPluralDef->le impériaux
  asker_faction_pawnsPluralIndef->un impériaux
  asker_faction_leaderTitle->haut gouverneur stellaire
  asker_faction_royalFavorLabel->faveur royale

```
